src/utils/user_registry.ts

```diff
@@ -44,9 +45,16 @@ export const blurUser = (
       )
     ),
     TE.chainW((res) => {
-      switch (res.status) {
+      const status = res.status;
+      const value = res.value;
+      switch (status) {
         case 200:
-          return TE.of(some({ id: res.value.id }));
+          return pipe(
+            value,
+            O.fromNullable,
+            O.map(responeValue => some({ id: responeValue.id })),
+            O.fold(()=>TE.left<ErrorResponses>(ResponseErrorInternal("Error reading response data")), (data) => TE.of(data))
```

Comment:
When you need to convert a data type into another, use _fromXYZ_ smart constructor. For example, `TE.fromOption` does exactly what you expect:
```suggestion
            TE.fromOption(() => ResponseErrorInternal("Error reading response data"))
```
```

Reply:
@balanza  resolved by https://github.com/pagopa/hub-spid-login-ms/pull/52/commits/f8414cca57d92c4a6f623ff7d4667a69b455a046